### DISCUSSION TITLE
Retrieving values from OBS

### DISCUSSION THREAD

**Discussion by mccombey** [obs-websocket-community-projects/obs-websocket-js#280] [Q&A]
I've been successfully controlling OBS using obs-websocket-js and it works fantastically well. However, I have no idea how to retrieve values from OBS using code. I've looked everywhere and there are no examples of syntax. Functions like GetTextGDIPlusProperties are there, but how do I use them?

This is how I'm currently using the obs-websocket-js to send commands...

```
const obs = new OBSWebSocket();
	obs.connect({
		address: "localhost:4444"
	});

obs.send("SetCurrentScene", {
      "scene-name": "titles"		
});	
```

Any help would be most appreciated!

**Comment by t2t2** [comment]
`.send` function calls return a [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) that resolve when obs-websocket answers to the request. Therefore the data can be accessed by

`.then` on promise chain:

```ts
// SetCurrentScene returns an empty object on resolve, so while you can use the callback to know when scene has been set, Get works as a better example:
obs.send("GetCurrentScene").then((data) => {
	console.log(data.name); // Current scene name
});
```

Alternatively there's also [async/await syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function):

```ts
// Generally should be in a async function like
// async function checkCurrentScene() {
const data = await obs.send("GetCurrentScene");

console.log(data.name); // Current scene name
```
